src/TumblThree/TumblThree.Applications/Crawler/AbstractTumblrCrawler.cs

```diff
@@ -227,18 +227,50 @@ protected async Task AddImgurAlbumUrlAsync(string post, string timestamp)
 
         protected void AddTumblrPhotoUrl(string post, int? postTimestamp)
         {
+            List<String> idCache = new List<string>();
+
             foreach (string imageUrl in TumblrParser.SearchForTumblrPhotoUrl(post))
             {
                 string url = imageUrl;
                 if (CheckIfSkipGif(url)) { continue; }
 
+                var matchesNewFormat = Regex.Match(url, "media.tumblr.com([A-Za-z0-9_/:.-]*)/s[0-9]*x[0-9]*");
+                if (matchesNewFormat.Success)
+                {
+                    string id = matchesNewFormat.Groups[1].Value;
+                    if (!idCache.Contains(id))
+                    {
+                        url = ResizeNewStyleTumblrImageUrl(url);
+                        idCache.Add(id);
+                    }
+                    else continue;
+                }
+
                 url = ResizeTumblrImageUrl(url);
                 url = RetrieveOriginalImageUrl(url, 2000, 3000);
+                
                 // TODO: postID
                 AddToDownloadList(new PhotoPost(url, Guid.NewGuid().ToString("N"), postTimestamp?.ToString(), BuildFileName(url, (Post)null, -1)));
             }
         }
 
+        protected string ResizeNewStyleTumblrImageUrl(string imageUrl)
+        {
+            var rgx = Regex.Match(imageUrl, "([A-Za-z0-9_/:.-])*/s([0-9]*)x([0-9]*)([A-Za-z0-9_/:.-])*");
+            int width = int.Parse(rgx.Groups[2].Value);
+            int height = int.Parse(rgx.Groups[3].Value);
+
+            const int maxRes = 1280 * 1920;
+
+            double widthProportion = width / (width * height);
```

Comment:
```suggestion
            double widthProportion = (double)width / (width * height);
```
One variable has to be casted, otherwise the result is always zero. Are you sure the algorithm works?

Reply:
Cheers for the correction. I don't think this original idea gave me the correct numbers anyway. I made a new mathematically correct version, however it turns out the tumblr url was not as malleable as I first thought. I've come up with a new robust idea which will involve caching the urls and looping through that cache afterwards to pick the most high-res option.